### ISSUE TITLE
Fix now outdated references to mesh files

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#1478]
#1453 moved a lot of tests but didn't fix all references to files in `tests/deal.II`. For example, `distributed/2d_refinement_03.debug` currently fails for me because it can't read this mesh file any more:

```
template<int dim>
void test(std::ostream & /*out*/)
{
  parallel::distributed::Triangulation<dim> tr(MPI_COMM_WORLD);
  Triangulation<dim> tr2 (Triangulation<dim>::limit_level_difference_at_vertices);

  {
    GridIn<dim> gi;
    gi.attach_triangulation (tr);
    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_02/2d.xda");
    try
      {
=>      gi.read_xda (in);
```

I find this:
```
1/dealii> grep /deal.II tests/*/*cc | grep SOURCE_DIR
tests/distributed_grids/2d_coarse_grid_02.cc:  std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_02/2d.xda");
tests/distributed_grids/2d_coarse_grid_03.cc:  std::ifstream in (SOURCE_DIR "/../deal.II/grid_in/2d.inp");
tests/distributed_grids/2d_coarsening_03.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_02/2d.xda");
tests/distributed_grids/2d_coarsening_03.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_02/2d.xda");
tests/distributed_grids/2d_refinement_03.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_02/2d.xda");
tests/distributed_grids/2d_refinement_03.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_02/2d.xda");
tests/distributed_grids/3d_coarse_grid_02.cc:  test<3> (SOURCE_DIR "/../deal.II/grid_in_3d/1.in");
tests/distributed_grids/3d_coarse_grid_02.cc:  test<3> (SOURCE_DIR "/../deal.II/grid_in_3d/2.in");
tests/distributed_grids/3d_coarse_grid_02.cc:  test<3> (SOURCE_DIR "/../deal.II/grid_in_3d/3.in");
tests/distributed_grids/3d_coarse_grid_02.cc:  test<3> (SOURCE_DIR "/../deal.II/grid_in_3d/4.in");
tests/distributed_grids/3d_coarse_grid_02.cc:  test<3> (SOURCE_DIR "/../deal.II/grid_in_3d/evil_0.in");
tests/distributed_grids/3d_coarse_grid_02.cc:  test<3> (SOURCE_DIR "/../deal.II/grid_in_3d/evil_1.in");
tests/distributed_grids/3d_coarse_grid_02.cc:  test<3> (SOURCE_DIR "/../deal.II/grid_in_3d/evil_2.in");
tests/distributed_grids/3d_coarse_grid_02.cc:  test<3> (SOURCE_DIR "/../deal.II/grid_in_3d/evil_3.in");
tests/distributed_grids/3d_coarse_grid_02.cc:  test<3> (SOURCE_DIR "/../deal.II/grid_in_3d/evil_4.in");
tests/distributed_grids/3d_coarse_grid_06.cc:  gi.read (SOURCE_DIR "/../deal.II/grid_in_3d_02/747.ucd");
tests/distributed_grids/3d_coarsening_03.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_3d/4.in");
tests/distributed_grids/3d_coarsening_03.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_3d/4.in");
tests/distributed_grids/3d_coarsening_04.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_3d_02/747.ucd");
tests/distributed_grids/3d_coarsening_04.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_3d_02/747.ucd");
tests/distributed_grids/3d_refinement_03.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_3d/4.in");
tests/distributed_grids/3d_refinement_03.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_3d/4.in");
tests/distributed_grids/3d_refinement_08.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_3d_02/747.ucd");
tests/distributed_grids/3d_refinement_08.cc:    std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_3d_02/747.ucd");
tests/mpi/p4est_2d_coarse_01.cc:      std::ifstream in (SOURCE_DIR "/../deal.II/grid_in_02/2d.xda");
```
I believe that they all currently fail for me: http://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&buildid=11842

@tamiko  -- do you still have the map that describes where all of these mesh files were moved? How did that not show up in your testing?

**Comment by tamiko** [comment]
@bangerth I tested for all the categories into which I moved tests. I was not aware of the possibility of files under ```distributed_grids/...``` to read in files from ```deal.II```.

**Comment by tamiko** [comment]
@bangerth Fix and a fix for the mpi tests incoming. (The sad thing is that I spent roughly 1.5 hours testing the changes and still missed those bugs.)

**Comment by tamiko** [comment]
Closed by #1480
